### PR TITLE
Use UTF-8 encoding when reading pyproject.toml

### PR DESCRIPTION
**Pull request by domdfcoding**
This avoids a `UnicodeDecodeError` when reading `pyproject.toml` files containing unicode characters on Windows. These characters may be present in a project's metadata, such as its description or an author's name.

```python
>>> from pep517.check import check
>>> check(".")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "lib\site-packages\pep517\check.py", line 145, in check
    pyproject_data = toml_load(f)
  File "lib\site-packages\toml\decoder.py", line 156, in load
    return loads(f.read(), _dict, decoder)
  File "lib\encodings\cp1252.py", line 23, in decode
    return codecs.charmap_decode(input,self.errors,decoding_table)[0]
UnicodeDecodeError: 'charmap' codec can't decode byte 0x90 in position 242: character maps to <undefined>
```

Furthermore, the [TOML specification](https://toml.io/en/v1.0.0#spec) requires files to be encoded in UTF-8.

### CODE REVIEW

File: pep517/build.py [hunk 1]
```diff
@@ -31,7 +32,7 @@ def load_system(source_dir):
     Load the build system from a source dir (pyproject.toml).
     """
     pyproject = os.path.join(source_dir, 'pyproject.toml')
-    with open(pyproject) as f:
+    with io.open(pyproject, encoding="utf-8") as f:
```
Comment (domdfcoding) [inline pep517/build.py:35]: `io.open` is necessary for Python 2, where `builtins.open` lacks an "encoding" argument.

**Comment by takluyver** [comment]
Just as a reminder, `envbuild` and all the pieces that depend on it, including `pep517.build` are deprecated (see #91).

This is a simple enough improvement that I don't mind merging it if you want to resolve the conflicts, though.

**Comment by domdfcoding** [comment]
> This is a simple enough improvement that I don't mind merging it if you want to resolve the conflicts, though.

Done

**Comment by takluyver** [comment]
Thanks!

**Pull request merged**